pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java

```diff
@@ -54,7 +54,7 @@ public void init(@Nullable Set<String> fields, RecordExtractorConfig recordExtra
   }
 
   private Object getFieldValue(Descriptors.FieldDescriptor fieldDescriptor, Message message) {
-    if (fieldDescriptor.isRepeated() || !fieldDescriptor.hasOptionalKeyword() || message.hasField(fieldDescriptor)) {
+    if (fieldDescriptor.isRepeated() || !fieldDescriptor.isOptional() || message.hasField(fieldDescriptor)) {
```

Comment:
Are strings and bytes marked as isRepeated by default? Reading the proto documentation it seems we cannot distinguish between null and empty value in strings and bytes.

Reply:
isRepeated is only for collections. We dont want to have the null behavior for collections.

Reply:
> Are strings and bytes marked as isRepeated by default? Reading the proto documentation it seems we cannot distinguish between null and empty value in strings and bytes.

isRepeated is only for collections. We dont want to have the null behavior for collections.